test/test_utils/ec2.py

```diff
@@ -105,43 +108,43 @@ def get_efa_ec2_instance_type(default, filter_function=lambda x: x, job_type="")
 
 
 def get_ec2_instance_type(
-    default, processor, filter_function=lambda x: x, efa=False, arch_type="", job_type=""
+    default, processor, filter_function=lambda x: x, arch_type="", job_type=""
 ):
     """
-    Get EC2 instance type from associated EC2_[CPU|GPU]_INSTANCE_TYPE env variable, or set it to a default
-    for contexts where the variable is not present (i.e. PR, Nightly, local testing)
+    Get EC2 instance type from associated EC2_[CPU|GPU]_INSTANCE_TYPE env variable, or set it to a
+    default for contexts where the variable is not present (i.e. PR, Nightly, local testing)
 
-    :param default: Default instance type to use - Should never be p3dn
+    :param default: Default instance type to use
     :param processor: "cpu" or "gpu"
-    :param filter_function: filter_function(instance_type_list) A function that takes the list to be generated by
-    the logic of the get_ec2_instance_type function, and filters the list to only produce "acceptable" instances.
-    For example, this can be a function that only returns multi-gpu instance types from a given list of instance types.
+    :param filter_function: filter_function(instance_type_list) A function that takes the list to be
+    generated by the logic of the get_ec2_instance_type function, and filters the list to only
+    produce "acceptable" instances. For example, this can be a function that only returns multi-gpu
+    instance types from a given list of instance types.
 
-    :return: one item list of instance type -- this is used to parametrize tests, and parameter is required to be
-    a list.
+    :return: one item list of instance type -- this is used to parametrize tests, and parameter is
+    required to be a list.
     """
+    if is_pr_context():
```

Comment:
In what scenario would we be on an instance type that has efa interfaces but efa dedicated env variable is not truthy?

Reply:
Since we parametrize instances for tests on PRs at the test-function-level, we need a way to say "I want this test to always run on p4d.24xlarge. However, I don't want it to run all the time on all CI job executions. Instead, I'll enable the `ec2_efa_tests` configuration when I'm contributing something relevant, which is when this test should run on PRs."

https://github.com/aws/deep-learning-containers/blob/428d7539d727e96253887e7738f67e13cb3c9ed8/dlc_developer_config.toml#L55-L57

So when `ec2_efa_tests` is `False`, `EFA_DEDICATED` is also `False`, preventing this test from running at the parametrization level, which prevents "heavy" tests from running on PRs due to human error.

This would work better than the previous system, where the code-owner was meant to supply both the "p4d.24xlarge" instance-type as well as `efa=True` as a parameter in the same function call. This doesn't really work, because the entity that decides whether instance-types should be parametrized is not the code itself when it is written, but rather the configuration of the test job and the tests at runtime.